src/main/java/com/example/books/dto/CreateBookRequestDto.java

```diff
@@ -1,13 +1,20 @@
 package com.example.books.dto;
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
+    @NotNull
     private String author;
+    @NotNull
     private String isbn;
+    @NotNull
+    @Min(0)
     private BigDecimal price;

```

Comment:
Poor validation make it more custom

Reply:
And how exactly it should look like? Maybe you should add some description of the fields, so I would know what constraints it should have?

Reply:
There is no any prescribed approach. We use either technical requirements or common sense.
Most used predefined annotations are '@NotNull', '@NotEmpty', '@NotBlank', '@Length'.
Here '@Isbn' would also be very natural